### PR TITLE
Move the QOs to TRFC conversion to a class, so it can be tested

### PR DESCRIPTION
**Pull request by knopers8**
Needs https://github.com/AliceO2Group/QualityControl/pull/673

### CODE REVIEW

**Comment by Barthelemy** [comment]
the build errors are legit

**Comment by knopers8** [comment]
Oh, indeed. I forgot that I need a PR to O2 first.

**Comment by knopers8** [comment]
It'll go after this https://github.com/AliceO2Group/AliceO2/pull/5941

File: Framework/include/QualityControl/QualitiesToTRFCollectionConverter.h [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+/// \file   QualitiesToTRFCollectionConverter.h
+/// \author Piotr Konopka
+///
+
+#ifndef QUALITYCONTROL_QUALITIESTOTRFCOLLECTIONCONVERTER_H
+#define QUALITYCONTROL_QUALITIESTOTRFCOLLECTIONCONVERTER_H
+
+#include "QualityControl/QualityObject.h"
+#include <DataFormatsQualityControl/FlagReasons.h>
+#include <DataFormatsQualityControl/TimeRangeFlagCollection.h>
```
Comment (Barthelemy) [inline Framework/include/QualityControl/QualitiesToTRFCollectionConverter.h:21]: I think but am not certain that you could forward declare it. Also, is the one above necessary ?

File: Framework/src/QualitiesToTRFCollectionConverter.cxx [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+
+  // Is the beginning of time range covered by the first QO provided?
+  if (mCurrentStartTime < mStartTimeLimit && validFrom > mStartTimeLimit) {
+    mConverted->insert({ mStartTimeLimit, validFrom - 1, FlagReasonFactory::MissingQualityObject(), noQualityObjectsComment, mQOPath });
+  }
+
+  mCurrentStartTime = validFrom < mStartTimeLimit ? mStartTimeLimit : validFrom;
+  mCurrentEndTime = validUntil > mEndTimeLimit ? mEndTimeLimit : validUntil;
+
+  if (!mCurrentTRF.has_value() && newQO.getQuality().isWorseThan(Quality::Good)) { //todo check if no QO in some period before the first one arrives
```
Comment (Barthelemy) [inline Framework/src/QualitiesToTRFCollectionConverter.cxx:64]: I don't really undesrtand the todo.

**Review by Barthelemy — APPROVED** [review]
Clean and very clear code. Just 2 questions but nothing blocking at all.

File: Framework/src/QualitiesToTRFCollectionConverter.cxx (hunk 2 above)
Comment (knopers8) [reply Framework/src/QualitiesToTRFCollectionConverter.cxx:64]: this is my bad habit of forgetting to remove "todo"s after they are done.

File: Framework/include/QualityControl/QualitiesToTRFCollectionConverter.h (hunk 1 above)
Comment (knopers8) [reply Framework/include/QualityControl/QualitiesToTRFCollectionConverter.h:21]: you're right, i'll fix it.

**Pull request merged**